realm/realm-jni/build.gradle

```diff
@@ -87,42 +82,22 @@ if (project.forceDownloadCore instanceof String) {
     ext.forceDownloadCore = Boolean.parseBoolean(project.forceDownloadCore)
 }
 
-def getNdkDir() {
-    def ndkDir = null
-    if (project.hasProperty('ndk.dir')) {
-        ndkDir = project.ext['ndk.dir']
-    } else if (System.getenv('NDK_HOME')) {
-        ndkDir = System.getenv('NDK_HOME')
-    }
-    if (ndkDir) {
-        if (ndkDir[-1] == "/") {
-            ndkDir = ndkDir[0..-2]
-        }
-    } else {
-        throw new GradleException('The ndk.dir property in the local.properties file or NDK_HOME is not set. One of them should be set.')
-    }
-    if (!file(ndkDir).directory) {
-        throw new GradleException('The path provided in the ndk.dir property in the local.properties or NDK_HOME is not a folder.')
+def getNdk() {
+    if (!System.env.NDK_HOME) {
+        throw new GradleException('The NDK_HOME environment variable is not set.')
     }
-   return ndkDir
-}
-
-def verifyNdk(ndkDir) {
-    def releaseFile = file("${ndkDir}/RELEASE.TXT")
-    if (!releaseFile.file) {
-        throw new GradleException('The path provided in the ndk.dir property in the local.properties or NDK_HOME does not seem to be an Android NDK.')
+    def ndkDir = file(System.env.NDK_HOME)
+    if (!ndkDir.directory) {
+        throw new GradleException('The path provided in the NDK_HOME environment variable is not a folder.')
     }
-    def ndkVersion = releaseFile.getText('UTF-8').split()[0]
-    if (ndkVersion != properNdkVersion) {
-        throw new GradleException("Your NDK version: ${ndkVersion}. Realm JNI should be compiled with the version ${properNdkVersion} of NDK.")
+    if (!file("${ndkDir}/RELEASE.TXT").file) {
+        throw new GradleException('The path provided in the NDK_HOME environment variable does not seem to be an Android NDK.')
     }
-}
-
-task checkProperties(group: 'check', description: 'Check the user provided gradle.properties') << {
-    verifyNdk(getNdkDir())
-    if (project.ext.has('encryption') && project.ext['encryption'] == 'true') {
-        project.coreVersion += '-encryption'
+    def detectedNdkVersion = file("${ndkDir}/RELEASE.TXT").text.trim().split()[0]
+    if (detectedNdkVersion != ndkVersion) {
+        throw new GradleException("Your NDK version: ${detectedNdkVersion}. Realm JNI should be compiled with the version ${ndkVersion} of NDK.")
     }
+    return ndkDir
 }
 
 ext.coreArchiveFile = project.file("../core-android-${project.coreVersion}.tar.gz")
```

Comment:
I think it's better not to move the location of core archive file.

To keep the location, `ext.coreArchiveFile` should be `rootProject.file("core-android-${project.coreVersion}.tar.gz")`